### ISSUE TITLE
Sub-lists are not checked for the `list-item` rule

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/awesome-lint#57]
This readme passes `awesome-lint`: https://github.com/juliandavidmr/awesome-nestjs/blob/a0c49d8dc6afa67a0c64f332fd59daccfe5b208f/README.md#resources But a lot of the descriptions are missing a `.` at the end.

We should ensure the other custom rules work with nested list items too.

// @transitive-bullshit

**Comment by IssuehuntBot** [comment]
@issuehunt has funded $40.00 to this issue.

---
- Submit pull request via [IssueHunt](https://issuehunt.io/repos/64659785/issues/57) to receive this reward.
- Want to contribute? Chip in to this issue via [IssueHunt](https://issuehunt.io/repos/64659785/issues/57).
- Checkout the [IssueHunt Issue Explorer](https://issuehunt.io/issues) to see more funded issues.
- Need help from developers? [Add your repository](https://issuehunt.io/users/44827199/repositories) on IssueHunt to raise funds.

**Comment by itaisteinherz** [comment]
I'll take a shot at fixing this 👍🏻

**Comment by transitive-bullshit** [comment]
Thanks @itaisteinherz 😄

afaik sub list-items should definitely be checked the same as normal list-items. I believe the problem here was #44 since tabbed sub-lists aren't parsed by remark as sub-lists, but rather additional plaintext on the original list-item.

**Comment by itaisteinherz** [comment]
Sorry for the delay with this, I've been very busy lately. I'll probably open a PR which fixes this within a week or so 🤞🏻

**Comment by itaisteinherz** [comment]
_Just sharing what I found after tinkering around with `awesome-lint` and `awesome-nestjs`:_

Given the following readme:

```md
# Awesome Nest

## Contents

- [Resources](#resources)

## Resources

- Tutorials
  - [Nestjs30Days](https://github.com/m24927605/Nestjs30Days) - Learn the Nestjs framework in 30 days
```

`awesome/list-item` won't report any errors. However, if we remove the `Contents` section:

```md
# Awesome Nest

## Resources

- Tutorials
  - [Nestjs30Days](https://github.com/m24927605/Nestjs30Days) - Learn the Nestjs framework in 30 days
```

The rule will report an error:

```
  ✖  6:62  List item description must end with proper punctuation       remark-lint:awesome/list-item
```

This seems to be what's causing the rest of the errors to not be reported as well.

**Comment by itaisteinherz** [comment]
I'm pretty sure this is the origin of the issue:

https://github.com/sindresorhus/awesome-lint/blob/dff8bb72dde17c0c6c51a1c8bbc8e86e6574a7c9/rules/list-item.js#L68

**Comment by itaisteinherz** [comment]
Opened an issue in [`unist-util-find-all-after`](https://github.com/syntax-tree/unist-util-find-all-after): syntax-tree/unist-util-find-all-after#5.

**Comment by IssuehuntBot** [comment]
@sindresorhus has rewarded $36.00 to @itaisteinherz. [See it on IssueHunt](https://issuehunt.io/repos/64659785/issues/57)

- :moneybag: Total deposit: $40.00
- :tada: Repository reward(0%): $0.00
- :wrench: Service fee(10%): $4.00
